TestRunners/TestResult.py

```diff
@@ -0,0 +1,81 @@
+from .Utils import TERMINAL_TEXT_STYLES
+
+
+class TestResult:
+    """Used to generate a string representing a test result, detailing 
+    the test name, if it passed, and any additional notes. 
+    test_has_passed, test_has_failed, or test_is_ignored must be 
+    called at least once for a string to generate without error. 
+
+    """
+
+    __TEST_STATUS = {
+        "UNSET": None,
+        "IGNORED": "IGNORED",
+        "PASSED": f"{TERMINAL_TEXT_STYLES['GREEN_COLOR']}PASSED{TERMINAL_TEXT_STYLES['RESET_COLOR']}",
+        "FAILED": f"{TERMINAL_TEXT_STYLES['RED_COLOR']}FAILED{TERMINAL_TEXT_STYLES['RESET_COLOR']}",
+    }
+
+    def __init__(self, file_name: str):
+        self.__file_name = file_name
+        self.__test_notes: list[str] = []
+        self.__test_status = self.__TEST_STATUS["UNSET"]
+        self.__object_type = "File"
+
+    def set_object_type_to_folder(self):
+        """When to_string is called, test will be referred to as a folder 
+        """
+        self.__object_type = "Folder"
+
+    def set_object_type_to_file(self):
+        """"When to_string is called, test will be referred to as a file. 
+        This is the default behavior and does not need to be called if the test 
+        is intended to represent a file and not a folder. 
+        """
+        self.__object_type = "File"
+
+    def test_has_passed(self, note=""):
```

Comment:
I find the naming convention you've chosen in this file confusing, because `test_has_failed` sounds like a bool.

I suggest the following names:
* `has_passed()`/`has_failed()` for the booleans (i.e., the methods that now start with `check...`
* `pass()`/`fail()` for the void operations that set the test output status (this matches the standard xUnit terminology, as a bonus)

Reply:
I wish I could use `pass` but that's a special keyword in Python, so I'll set it to `pass_test` and `fail_test` + changing bool methods to `has_passed` and `has_failed`